src/lib.rs

```diff
@@ -151,38 +150,52 @@ pub struct GptDisk {
 impl GptDisk {
     /// Add another partition to this disk.  This tries to find
     /// the optimum partition location with the lowest block device.
+    /// Returns the new partition id if there was sufficient room
+    /// to add the partition. Size is specified in bytes.
     pub fn add_partition(
         &mut self,
         name: &str,
         size: usize,
         part_type: partition_types::Type,
         flags: u64,
-    ) -> io::Result<()> {
+    ) -> io::Result<u32> {
+        let size_lba = match size.checked_div(self.config.lb_size.into()) {
```

Comment:
I know we talked about this already, but you could probably just do a self.config.lb_size < logical block size, though at least checked_div is a quick match instead of an if.

Reply:
Yeah I know.  I still need the result of that calculation though for later.  If not then yeah I'd def just do a < comparison.